src/NRediSearch/Suggestion.cs

```diff
@@ -0,0 +1,108 @@
+﻿// .NET port of https://github.com/RedisLabs/JRediSearch/
+using System;
+using StackExchange.Redis;
+
+namespace NRediSearch
+{
+    public class Suggestion
+    {
+        public string String { get; }
+        public double Score { get; }
+        public string Payload { get; }
+
+        private Suggestion(SuggestionBuilder builder)
+        {
+            String = builder._string;
+            Score = builder._score;
+            Payload = builder._payload;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (this == obj)
+            {
+                return true;
+            }
+
+            if (obj == null || obj.GetType() != GetType())
```

Comment:
prefer something like `if (!(obj is Suggestion that)) return false;`, and remove the `that` from 2 lines below

Reply:
Nice!